### ISSUE TITLE
Poor colour contrast in form elements while connecting Jetpack

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#31550]
While connecting Jetpack site from WordPress.com, I see this:

<img width="427" alt="Screenshot 2019-03-19 at 14 13 07" src="https://user-images.githubusercontent.com/87168/54605369-c89c3e80-4a51-11e9-889a-cc478ed92038.png">

_"Setting up"_ button seems like I should see that text even if the button is disabled and in "loading" state; currently it has too low colour contrast even for my healthy eyes and good screen. :-) In other places in this flow we pulsate between visible and grey text when something is happening (sorry no screengrabs about this).

The address in disabled input is also pretty low contrasty.

Later in the flow also these inputs could have better contrast:

<img width="505" alt="Screenshot 2019-03-19 at 14 13 51" src="https://user-images.githubusercontent.com/87168/54605368-c89c3e80-4a51-11e9-98ce-cb5858d3bcb0.png">

This is really just generic issue with form buttons (see https://github.com/Automattic/wp-calypso/issues/26982) but I feel that in this flow we could work around the issue of using disabled form elements in some way.

FYI @tyxla @keoshi

**Comment by simison** [comment]
https://webaim.org/resources/contrastchecker/ is excellent too to check these pass.

**Comment by tyxla** [comment]
Thanks @simison! What could we do about these @keoshi @jeffgolenski?

**Comment by simison** [comment]
Especially this note by @drw158:

> Disabled inputs are being used to convey information that is meant to be read. Important info should not be placed in a disabled input because disabled elements are not meant to be legible.

via https://github.com/Automattic/wp-calypso/issues/26982#issuecomment-474396291

**Comment by keoshi** [comment]
Thanks for bringing this up @simison !

> This is really just generic issue with form buttons (see #26982)

I agree that we should take care of this higher up the chain and benefit from the changes downstream.

> but I feel that in this flow we could work around the issue of using disabled form elements in some way.

I'm thinking we could do without that URL form once a valid URL has been submitted and have a more visual UI for that intermediate step, but that also feels more complex than what we're able to handle right now. What did you have in mind?

**Comment by simison** [comment]
> > but I feel that in this flow we could work around the issue of using disabled form elements in some way.
>
>I'm thinking we could do without that URL form once a valid URL has been submitted and have a more visual UI for that intermediate step, but that also feels more complex than what we're able to handle right now. What did you have in mind?

Here's a recording of the flow I took: https://cloudup.com/czVZnjUpym3

I'd imagine the part with "setting up" and URL input could also be part of some other flow, so I don't actually see the full picture here?

<img width="439" alt="image" src="https://user-images.githubusercontent.com/87168/54633384-be496700-4a88-11e9-89c1-8a86cda9c54d.png">

Otherwise, the flow could just directly jump from this:

<img width="729" alt="image" src="https://user-images.githubusercontent.com/87168/54633357-af62b480-4a88-11e9-83b8-f736e86f9d95.png">

To something that looks like this:

<img width="424" alt="image" src="https://user-images.githubusercontent.com/87168/54633402-c73a3880-4a88-11e9-8f3c-02783dcd4eda.png">

Currently, there are two visually different screens (and a quick flash of W logo, so three) all basically telling "please wait". 

If the URL input + button step has to stay, perhaps instead of disabled button you could just show the text, perhaps with rolling animation like in URL input?

![image](https://user-images.githubusercontent.com/87168/54634207-657ace00-4a8a-11e9-992c-a60420576123.png)

URL input also could dissapear at that point and just show the URL as a text.

**Comment by simison** [comment]
Here's an example from Mailchimp signup form:

<img width="224" alt="image" src="https://user-images.githubusercontent.com/87168/54635867-d7a0e200-4a8d-11e9-9a20-b690040fb910.png">

https://login.mailchimp.com/signup/

First disabled state is when we don't really need to know what it says on the button and we haven't yet filled the form.

Then we've filled the form.

And finally we're waiting for form to send. There's a spinning animation.

**Comment by keoshi** [comment]
The improvements on the flow all look really good and we're considering some on the Figma file I liked to on Slack, but they're out of the scope of this issue so let's create a new one for that.

As for the pattern example from Mailchimp, that looks interesting and is certainly something we could champion and push the improvements upstream! Two notes:

- I've checked their code and it doesn't seem optimal though, as they're not disabling the button allowing for sequential requests (not sure if that would fly in JPC).
- Maybe we can accomplish the same effect by adding an extra class to the last disabled state and override the styles then.

**Comment by keoshi** [comment]
> Maybe we can accomplish the same effect by adding an extra class to the last disabled state and override the styles then.

Here's a quick prototype showing that:

https://codepen.io/keoshi/pen/bZxPrP

**Comment by keoshi** [comment]
I'll be looking at this in the upcoming HACK week.

**Comment by jeffgolenski** [comment]
You're a champion, @keoshi

**Comment by keoshi** [comment]
Created a PR to sort this issue here: https://github.com/Automattic/wp-calypso/pull/50849
